sky/services/dynamic/dynamic_service.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_SERVICES_DYNAMIC_DYNAMIC_SERVICE_H_
+#define SKY_SERVICES_DYNAMIC_DYNAMIC_SERVICE_H_
+
+#include "sky/services/dynamic/dynamic_service_macros.h"
+#include "mojo/public/c/system/types.h"
+#include "mojo/public/c/environment/async_waiter.h"
+#include "mojo/public/c/environment/logger.h"
+
+FLUTTER_C_API_START
+
+FLUTTER_EXPORT void FlutterServiceOnLoad(const MojoAsyncWaiter* waiter,
```

Comment:
this isn't valid C. in C, you would have to spell this type name

const struct MojoAsyncWaiter* waiter

and write (void) instead of () in the Unload declaration, etc. if you want to verify that these things compile as C (and not just as C++ in an extern "C" declaration) you should probably have a .c file somewhere that #includes the headers you want to be C. or you could say that these APIs are only usable from C++ / Obj-C++, but that might be more annoying for consumers.